### DISCUSSION TITLE
How to get contents of the files with a specific name in a given repository?

### DISCUSSION THREAD

**Discussion by buremba** [community/community#24871] [Apps, API and Webhooks]
<p>I’m trying to fetch the contents of all the files with the name <code>_config.jsonnet</code> in this repository (<a href="https://github.com/rakam-io/segment-recipe" rel="nofollow noopener">https://github.com/rakam-io/segment-recipe</a>).</p>
<p>I tried the following query:</p>
<pre><code>query GetFilesQuery($branch: GitObjectID, $query: String!) {
  search(first: 1, type: REPOSITORY, query: $query) {
    edges {
      node {
        ... on Repository {
          object(expression: "master:", oid: $branch) {
            ... on Tree {
              entries {
                name
                object {
                  ... on Blob {
                    text
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>with the parameters with no luck:</p>
<pre><code>{"branch": null, "query": "repo:rakam-io/segment-recipe extension:jsonnet filename:_config"}
</code></pre>
<p>Is there anything that I’m missing or the API doesn’t support it&gt;</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>The GraphQL search connection currently doesn’t support code results, only repository, issue, and user results. For now, you’ll have to use <a href="https://developer.github.com/v3/search/#search-code" rel="nofollow noopener">the v3 API to search for code</a>.</p>
<p>I hope that helps!</p>

**Comment by diouldebalde** [comment]
<p>OK j’essaie pour voir le résultat…</p>
